origin/origin_api.go

```diff
@@ -57,3 +62,54 @@ func ConfigureXrootdMonitoringDir() error {
 
 	return nil
 }
+
+// Remove tmp files under ${Origin_RunLocation}/export/pelican/monitoring
+// from diector-based/self tests. There could be dangling files due to
+// error in testing
+func LaunchOriginFileTestMaintenance(ctx context.Context) {
+	monitoringDir := filepath.Join(param.Origin_RunLocation.GetString(), "export", "pelican", "monitoring")
+
+	server_utils.LaunchWatcherMaintenance(
+		ctx,
+		[]string{monitoringDir},
+		"director-based origin tests clean up",
+		1*time.Minute,
+		func(notifyEvent bool) error {
+			entries, err := os.ReadDir(monitoringDir)
+			dirPrevFile := ""
+			selfPrevFile := ""
+			if err != nil {
+				return err
+			}
+			for _, entry := range entries {
+				if !entry.IsDir() {
+					fn := entry.Name()
+					if strings.HasPrefix(fn, "director") {
+						// Always leave the first item
+						if dirPrevFile != "" {
```

Comment:
How do we know the first file is the most recent/the one that we want to keep for the test files?

Reply:
Ahh I'm sorry. What I meant was to leave the "last" item in the directory. This is because the `os.ReadDir` sorts the files, and for all the test files, we have a timestamp at the end of the file, which is string-comparable, and we leave the last item as the latest test file, so that there's won't be a race condition where the just-arrived test file was deleted.

Reply:
I'll revise the comment to be more self-explanatory.